iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -429,45 +434,50 @@ public boolean setColStatistics(org.apache.hadoop.hive.ql.metadata.Table hmsTabl
               ImmutableMap.of()));
       writer.finish();
     } catch (IOException e) {
-      LOG.error(String.valueOf(e));
+      LOG.error("Unable to write stats to puffin file", e.getMessage());
     }
     return false;
   }
 
   @Override
   public boolean canProvideColStatistics(org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
     Table table = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
-    if (canSetColStatistics(hmsTable)) {
-      Path statsPath = getStatsPath(table);
-      try {
-        FileSystem fs = statsPath.getFileSystem(conf);
-        if (fs.exists(statsPath)) {
-          return true;
-        }
-      } catch (IOException e) {
-        LOG.warn("Exception when trying to find Iceberg column stats for table:{} , snapshot:{} , " +
-            "statsPath: {} , stack trace: {}", table.name(), table.currentSnapshot(), statsPath, e);
+    return canSetColStatistics(hmsTable) && canProvideColStatistics(table, table.currentSnapshot().snapshotId());
+  }
+
+  private boolean canProvideColStatistics(Table table, long snapshotId) {
+    Path statsPath = getStatsPath(table, snapshotId);
+    try {
+      FileSystem fs = statsPath.getFileSystem(conf);
+      if (fs.exists(statsPath)) {
```

Comment:
simply return result of fs.exists(statsPath)

Reply:
fixed